Samples/Notifications/App/CppUnpackagedAppNotifications/CppUnpackagedAppNotifications/App.xaml.cpp

```diff
@@ -0,0 +1,101 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#include "pch.h"
+#include "App.xaml.h"
+#include "MainWindow.xaml.h"
+#include "Notifications\NotificationManager.h"
+#include <wil/result.h>
+#include <Microsoft.UI.Xaml.Window.h>
+#include <winrt/Microsoft.Windows.AppLifecycle.h>
+#include <winrt/Microsoft.Windows.AppNotifications.h>
+
+namespace winrt
+{
+    using namespace Windows::Foundation;
+    using namespace Microsoft::UI::Xaml;
+    using namespace winrt::Microsoft::Windows::AppLifecycle;
+    using namespace winrt::Microsoft::Windows::AppNotifications;
+}
+
+// NotificationManager is responsible for registering and unregistering the Sample for App Notifications as well as
+// dispatching actionned notifications to the appropriate scenario.
+// Registration will happen when Init() is called and Unregistration will happen when this
+// instance variable goes out of scope, i.e.: when the App is terminated.
+static NotificationManager g_notificationManager;
+
+namespace winrt::CppUnpackagedAppNotifications::implementation
+{
+    static App* app{ nullptr };
+
+    App::App()
+    {
+        InitializeComponent();
+        app = this;
+
+#if defined _DEBUG && !defined DISABLE_XAML_GENERATED_BREAK_ON_UNHANDLED_EXCEPTION
+        UnhandledException([](winrt::IInspectable const&, winrt::UnhandledExceptionEventArgs const& e)
+            {
+                if (IsDebuggerPresent())
+                {
+                    auto errorMessage = e.Message();
+                    __debugbreak();
+                }
+            });
+#endif
+    }
+
+    void App::ToForeground()
+    {
+        assert(app != nullptr);
+
+        HWND hwnd;
+        auto WindowNative{ app->window.as<IWindowNative>() };
+        if (WindowNative != nullptr && WindowNative->get_WindowHandle(&hwnd) == S_OK)
+        {
+            SwitchToThisWindow(hwnd, TRUE);
+        }
+    }
+
+    std::wstring App::GetFullPathToExe()
```

Comment:
Good place to start using r-value refs. So this method can return std::wstring&& and the caller could use std::move for assignment which would avoid a copy.

Reply:
In general, we could do this everywhere we utilize and return strings.